tests/ipfwd/test_mtu.py

```diff
@@ -7,12 +7,103 @@
 from datetime import datetime
 
 pytestmark = [
-    pytest.mark.topology('t1')
+    pytest.mark.topology('t1', 't2')
 ]
 
+'''
+In case of multi-dut we need src_host_ip, src_router_ip, dst_host_ip, src_ptf_port_list, dst_ptf_port_list for the dut under test, 
+to take care of that made changes in the testcase 
+'''
+
+def lag_facts(dut, mg_facts):
+    facts = {}
+
+    if not mg_facts['minigraph_portchannels']:
+        pytest.fail("minigraph_portchannels is not defined")
+
+    # minigraph facts
+    src_lag = mg_facts['minigraph_portchannel_interfaces'][0]['attachto']
+    dst_lag = mg_facts['minigraph_portchannel_interfaces'][2]['attachto']
+    logger.info("src_lag is {}, dst_lag is {}".format(src_lag, dst_lag))
+
+    for intf in mg_facts['minigraph_portchannel_interfaces']:
+        if intf['attachto'] == dst_lag:
+            addr = ip_address(unicode(intf['addr']))
+            if addr.version == 4:
+                facts['dst_router_ip'] = intf['addr']
+                facts['dst_host_ip'] = intf['peer_addr']
+        if intf['attachto'] == src_lag:
+            addr = ip_address(unicode(intf['addr']))
+            if addr.version == 4:
+                facts['src_router_ip'] = intf['addr']
+                facts['src_host_ip'] = intf['peer_addr']
+
+    facts['dst_port_ids'] = []
+    for intf in mg_facts['minigraph_portchannels'][dst_lag]['members']:
+        facts['dst_port_ids'].append(mg_facts['minigraph_ptf_indices'][intf])
+
+    facts['src_port_ids'] = []
+    for intf in mg_facts['minigraph_portchannels'][src_lag]['members']:
+        facts['src_port_ids'].append(mg_facts['minigraph_ptf_indices'][intf])
+
+    return facts
+
+
+def port_facts(dut, mg_facts):
+    facts = {}
+
+    if not mg_facts['minigraph_interfaces']:
+        pytest.fail("minigraph_interfaces is not defined.")
+
+    # minigraph facts
+    src_port = mg_facts['minigraph_interfaces'][0]['attachto']
+    dst_port = mg_facts['minigraph_interfaces'][2]['attachto']
+    logger.info("src_port is {}, dst_port is {}".format(src_port, dst_port))
```

Comment:
Some topology has disabled ports. it would be better to random choose 2 ports that are both admin up and oper up?

Reply:
Also, what if  mg_facts['minigraph_interfaces'] is empty or less than 3 members?